code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,308 @@
+/obj/machinery/deployable/mounted/attackby(obj/item/I, mob/user, params) //This handles reloading the gun, if its in acid cant touch it.
+	. = ..()
+
+	if(!ishuman(user))
+		return
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return
+
+	reload(user, I)
+
+
+///Reloads gun
+/obj/machinery/deployable/mounted/proc/reload(mob/user, ammo_magazine)
+	if(!istype(ammo_magazine, /obj/item/ammo_magazine))
+		return
+
+	var/obj/item/ammo_magazine/ammo = ammo_magazine
+	
+	if(istype(gun, ammo.gun_type))
+
+		if(ammo.current_rounds <= 0)
+			to_chat(user, "<span class='warning'>[ammo] is empty!</span>")
+			return
+
+		if(gun.current_mag)
+			gun.unload(user,0,1)
+			update_icon_state()
+
+		var/tac_reload_time = max(0.5 SECONDS, 1.5 SECONDS - user.skills.getRating("firearms") * 5)
+		if(!do_after(user, tac_reload_time, TRUE, src))
+			return
+	
+		gun.reload(user, ammo_magazine)
+		update_icon_state()
+
+///Sets the user as manning the internal gun
+/obj/machinery/deployable/mounted/on_set_interaction(mob/user)
+	operator = user
+
+	. = ..()
+
+	if(!gun)
+		CRASH("[src] has been deployed and attempted interaction with [user] without having a gun. This shouldn't happen.")
+	
+	
+
+	RegisterSignal(operator, COMSIG_MOB_MOUSEDOWN, .proc/start_fire)
+	gun.RegisterSignal(operator, COMSIG_MOB_MOUSEUP, /obj/item/weapon/gun/proc/stop_fire)
+	RegisterSignal(operator, COMSIG_MOB_MOUSEDRAG, .proc/change_target)
+
+	for(var/X in gun.actions)
+		var/datum/action/A = X
+		A.give_action(operator)
+	var/obj/screen/ammo/hud = operator.hud_used.ammo
+	gun.hud_enabled ? hud.add_hud(operator) : hud.remove_hud(operator)
+	hud.update_hud(operator)
+
+	gun.gun_user = operator
+
+	update_view(operator)
+
+///Begins the Firing Process
+/obj/machinery/deployable/mounted/proc/start_fire(datum/source, atom/object, turf/location, control, params)
+	SIGNAL_HANDLER
+
+	update_icon_state()
+
+	if(gun.gun_on_cooldown(operator))
+		return
+
+	if(QDELETED(object))
+		return
+
+	if(object == src) //Clicking the gun gets off of it.
+		var/mob/living/carbon/human/user = source
+		user.unset_interaction()
+		return
+
+	var/target = object
+
+	var/list/modifiers = params2list(params)
+	if(istype(target, /obj/screen))
+		if(!istype(target, /obj/screen/click_catcher))
+			return
+		//Happens when you click a black tile
+		target = params2turf(modifiers["screen-loc"], get_turf(operator), operator.client)
+
+	if(!can_fire(target))
+		return
+
+	gun.set_target(target)
+
+	if(modifiers["right"] || modifiers["middle"] || modifiers["shift"])
+		if(gun.active_attachable?.flags_attach_features & ATTACH_WEAPON)
+			gun.do_fire_attachment()
+		return
+	if(gun.gun_firemode == GUN_FIREMODE_SEMIAUTO)
+		if(!gun.Fire(TRUE) || gun.windup_checked == WEAPON_WINDUP_CHECKING)
+			return
+		gun.reset_fire()
+		return
+	gun.gun_user.client.mouse_pointer_icon = 'icons/effects/supplypod_target.dmi'
+	SEND_SIGNAL(gun, COMSIG_GUN_FIRE)
```

Comment:
ok so this is a huge copy of gun Fire() code, this is really hacky.

I'm pretty sure you can just call gun's proc